### PR TITLE
REF: avoid special-casing in agg_series

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/string_.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def tolist(self):
 
     @classmethod
     def _from_scalars(cls, scalars, dtype: DtypeObj) -> Self:
-        if lib.infer_dtype(scalars, skipna=True) != "string":
+        if lib.infer_dtype(scalars, skipna=True) not in ["string", "empty"]:
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/string_.py:267]: Yes.  The agg_series special-casing that this PR removes makes this change necessary

[the remark this answers is not in the document]
Comment (rhshadrach) [reply pandas/core/arrays/string_.py:267]: Ahh, I misread. I thought we were passing the result of `lib.infer_dtype` to `_from_sequence` below.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**